Mathlib/RingTheory/Polynomial/HilbertPoly.lean

```diff
@@ -117,11 +116,37 @@ lemma hilbertPoly_poly_zero (p : F[X]) : hilbertPoly p 0 = 0 := rfl
 lemma hilbertPoly_poly_succ (p : F[X]) (d : ℕ) :
     hilbertPoly p (d + 1) = ∑ i in p.support, (p.coeff i) • preHilbertPoly F d i := rfl
 
-variable (F) in
 lemma hilbertPoly_X_pow_succ (d k : ℕ) :
     hilbertPoly ((X : F[X]) ^ k) (d + 1) = preHilbertPoly F d k := by
   delta hilbertPoly; simp
 
+lemma hilbertPoly_add_nat (p q : F[X]) (d : ℕ) :
+    hilbertPoly (p + q) d = hilbertPoly p d + hilbertPoly q d := by
```

Comment:
I don't understand the purpose of `_nat` in the name.

Reply:
> I don't understand the purpose of `_nat` in the name.

The purpose of naming it in this way is to follow the style of the lemmas `Polynomial.hilbertPoly_zero_nat`, `Polynomial.hilbertPoly_poly_zero`, etc. (which are also in the file RingTheory/Polynomial/HilbertPoly.lean).

Observe that if we only name the above lemma as `Polynomial.hilbertPoly_add`, then if later we have some lemma
```
lemma Polynomial.hilbertPoly_? (p : F[X]) (d e : ℕ) :
    Polynomial.hilbertPoly p (d + e) = ... := ...
```
then the new lemma will also have the name `Polynomial.hilbertPoly_add`, which will cause some trouble.

If we name preserve the name `Polynomial.hilbertPoly_add_nat`, then when we need to have some lemma
```
lemma Polynomial.hilbertPoly_? (p : F[X]) (d e : ℕ) :
    Polynomial.hilbertPoly p (d + e) = ... := ...
```
we can name the new lemma as `Polynomial.hilbertPoly_poly_add`, so the new lemma won't be confused with the old lemma.

Reply:
I think that `hilbertPoly_add` is probably better. We can cross the `d + e` bridge when we come to it. We have `Polynomial.eval_add` for example, and `eval` takes two inputs.

Reply:
Oh but in fact this already happens here, because we have `_zero_nat` and `_poly_zero`. I think we need to ask for advice here about a good naming convention.

Reply:
`zero_apply` and `apply_zero`, maybe?

Reply:
> `zero_apply` and `apply_zero`, maybe?

And shall I change the declaration `Polynomial.hilbertPoly_poly_succ` into `Polynomial.hilbertPoly_apply_succ`, and `Polynomial.hilbertPoly_add_nat` into `Polynomial.hilbertPoly_add_apply`?

Reply:
It's probably best to be consistent, I guess.